kernel_test.go

```diff
@@ -15,6 +16,249 @@ const (
 	success = "\u2713"
 )
 
+const (
+	connectionKey = "a0436f6c-1916-498b-8eb9-e81ab9368e84"
+	sessionID     = "ba65a05c-106a-4799-9a94-7f5631bbe216"
+	transport     = "tcp"
+	ip            = "127.0.0.1"
+	shellPort     = 57503
+	iopubPort     = 40885
+)
+
+type testJupyterClient struct {
+	shellSocket *zmq.Socket
+	ioSocket    *zmq.Socket
+}
+
+// newTestJupyterClient creates and connects a fresh client to the kernel. Upon error, newTestJupyterClient
+// will Fail the test.
+func newTestJupyterClient(t *testing.T) (testJupyterClient, func()) {
+	t.Helper()
+
+	addrShell := fmt.Sprintf("%s://%s:%d", transport, ip, shellPort)
+	addrIO := fmt.Sprintf("%s://%s:%d", transport, ip, iopubPort)
+
+	// Prepare the shell socket.
+	shell, err := zmq.NewSocket(zmq.REQ)
+	if err != nil {
+		t.Fatal("NewSocket:", err)
```

Comment:
We should utilize the same format as in the main test here, to make sure the output is consistent:

```
t.Fatalf("\t%s Test case produced unexpected results.", failure)
```

I realize now that I didn't change these in my code before, and should have.  Thus, don't feel like you need to change them before merging.  Mostly putting this here as a reminder to myself.